### ISSUE TITLE
Multiselect attribute options wrong translation on Add Product page

### ISSUE THREAD

**Issue by RomanMichak** [magento/magento2#19388]
Attribute options in multiselect on the New Product page displayed with wrong translation. After a product is saved and form reloaded it showed correct translated options.

### Preconditions
Magento 2.3-develop
PHP 7.1.23
MySQL 5.7

### Steps to reproduce
1. Create additional store SE 
2. Create a test attribute with configuration:
 Type: Multiple Select 
 Code: test_attribute
 Options with translations by store:
admin | default | SE
-------|--------|------
test 1 | test 1 EN | test 1 SE
test 2 | test 2 EN | test 2 SE
![2018-11-24_14-55](https://user-images.githubusercontent.com/34232076/48968486-fd930a00-eff8-11e8-8419-454eb9922666.jpg)
3. Add test_attribute to default attribute set
4. Open Add product page on admin

### Expected result
1. To see test_attribute options to have in multiselect values: test 1, test 2 

### Actual result
1. Actually, the test attribute options have values: test 1 EN, test 2 EN
![2018-11-24_14-56](https://user-images.githubusercontent.com/34232076/48968505-27e4c780-eff9-11e8-9928-54fc00f3275e.jpg)

After a product is saved options displayed with correct translations 
![2018-11-24_14-58](https://user-images.githubusercontent.com/34232076/48968523-5e224700-eff9-11e8-80a7-a7c01bb596cb.jpg)

**Comment by magento-engcom-team** [comment]
Hi @RomanMichak. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@RomanMichak do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by RomanMichak** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @RomanMichak. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @RomanMichak, here is your [Magento instance](https://i-19388-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-19388-2-3-develop.instances.magento-community.engineering/admin](https://i-19388-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by RomanMichak** [comment]
Issue is confirmed on 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @novikor. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@novikor Thank you for verifying the issue.

Unfortunately, not enough information was provided to created internal ticket. Please consider adding the following:
- [x] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
@novikor Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96621`, `MAGETWO-96622` were created

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by shikhamis11** [comment]
#MM19IN

**Comment by henryktews** [comment]
Duplicate of #6507

**Comment by engcom-Bravo** [comment]
Hello @RomanMichak 

Thank you for contribution and collaboration!

We are not able to reproduce this issue on the latest `2.3-develop` branch anymore by the steps provided.

If you manage to reproduce this issue on the latest 2.3-develop please  reopen or create a new Issue with all required details according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines).
